aws_protocol_adaptor/device_client/aws_nvmsgbroker.c

```diff
@@ -142,11 +147,22 @@ NvDsMsgApiErrorType nvds_msgapi_disconnect(NvDsMsgApiHandle h_ptr)
 {
 	if ((h_ptr == NULL))
 	{
-		IOT_ERROR("Essensial args missing for function nvds_msgapi_disconnect\n");
+		IOT_ERROR("Essential args missing for function nvds_msgapi_disconnect\n");
 		return NVDS_MSGAPI_ERR;
 	}
 	IoT_Error_t rc = FAILURE;
 	AWS_IoT_Client *client = (AWS_IoT_Client *)h_ptr;
+	for (int i = 0; i < num_subscriptions; i++)
+	{
+		rc = aws_iot_mqtt_unsubscribe(client, subscribed_topics[i], strlen(subscribed_topics[i]));
+		if (SUCCESS != rc)
+		{
+			IOT_ERROR("Unable to unsubscribe, error: %d\n", rc);
+			return NVDS_MSGAPI_ERR;
```

Comment:
I think in this case it's probably better to simply continue after logging the error, as we want to ensure that the disconnection actually occurs.